### ISSUE TITLE
Path to ocamlrun overflow

### ISSUE THREAD

**Issue by scemama** [ocaml/ocaml#9330]
Hello,
when `ocamlrun` is in a very long path in the filesystem, for example 
```
/home/scemama/spack/opt/spack/linux-debian9-haswell/gcc-6.3.0/quantum-package-develop-grwklbbl43j5oz2ufirg54nm4j5rzclj/external/opam/4.09.0/bin/ocamlrun
```
bytecode executables don't run because the shebang line is too long. So I guess the creation of the bytecode executables should fail in these situations.

For more info on `#!` limitations, see
https://www.in-ulm.de/~mascheck/various/shebang/

**Comment by gasche** [comment]
It is always possible to run bytecode executables by passing them explicitly to the `ocamlrun` binary instead of relying on the shebang:

```
ocamlrun foo.byte
```

**Comment by scemama** [comment]
I know! but I spent the whole day trying to understand why installing OCaml via OPAM in one directory worked, while in another it did not. The only output I had was `exit-code 127` which is does not give much info to find the error.

You misunderstood my issue : I just wanted to alert the compiler developers of this problem and avoid others to be in the same situation as me in the future.

I believe that OCaml should detect that the executables it produces will not work and print a warning, or even replace the shebang line with `#!/usr/bin/env ocamlrun"`in such cases.
The type for the shebang line should be a limited-size string, and the lines that are too long should be rejected by construction.

**Comment by xavierleroy** [comment]
PR #8622 addresses this issue.  It should be fixed in OCaml release 4.10.0 and up.  Please reopen this issue if the problem is still observed in 4.10.0.
